lib/Data/Time/Calendar/Quarter.hs

```diff
@@ -85,6 +89,21 @@ instance Read Quarter where
         m <- readPrec
         return $ YearQuarter y m
 
+instance HasDays Quarter where
```

Comment:
@AshleyYakeley something like this perhaps? I still need to fix some imports, tests, and possibly add more instances for `HasDays`, but I just want to validate first if this approach looks better